flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/heap/HeapIntVector.java

```diff
@@ -47,4 +49,45 @@ public int getInt(int i) {
 			return dictionary.decodeToInt(dictionaryIds.vector[i]);
 		}
 	}
+
+	@Override
+	public void setInt(int i, int value) {
+		if (i >= vector.length) {
+			throw new RuntimeException();
+		}
+		vector[i] = value;
+	}
+
+	@Override
+	public void setIntsFromBinary(int rowId, int count, byte[] src, int srcIndex) {
+		if (LITTLE_ENDIAN) {
+			UNSAFE.copyMemory(src, BYTE_ARRAY_OFFSET + srcIndex, vector,
+					INT_ARRAY_OFFSET + rowId * 4L, count * 4L);
+		} else {
+			long srcOffset = srcIndex + BYTE_ARRAY_OFFSET;
+			for (int i = 0; i < count; ++i, srcOffset += 4) {
+				vector[i + rowId] = Integer.reverseBytes(UNSAFE.getInt(src, srcOffset));
+			}
+		}
+	}
+
+	@Override
+	public void setInts(int rowId, int count, int value) {
+		if (rowId + count - 1 >= vector.length) {
+			throw new RuntimeException();
```

Comment:
Add a meaningful message, or throw an index-out-of-bound exception?

Reply:
Nice catch!

Reply:
I think we can remove this check, leave the exception produced by Java array.